src/main/java/Duke.java

```diff
@@ -1,10 +1,124 @@
+import java.util.Scanner;
+
 public class Duke {
+    // constant for number of maximum tasks
+    public static final int MAXIMUM_TASKS = 100;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String userInput;
+        Scanner in = new Scanner(System.in);
+
+        Task[] taskList = new Task[MAXIMUM_TASKS];
+        int taskCount = 0;
+
+        printWelcomeMessage();
+
+        while (true) {
+            // take in user input from command line
+            userInput = in.nextLine();
+
+            if (userInput.equals("bye")) {
+                // if command "bye" is entered, stop asking for commands and exit
+                break;
+            } else if (userInput.equals("list")) {
+                // prints the task list if command "list" is given
+                printTaskList(taskList, taskCount);
+            } else if (userInput.contains("done")) {
+                // if input contains "done", find the corresponding task and update its status in the list
+                int donePosition = userInput.indexOf("done");
+                int doneTaskIndex = Integer.parseInt(userInput.substring(donePosition + 5));
+                if (doneTaskIndex > taskCount) {
+                    // if task indicated exceeds current task list, indicate an error
+                    printInvalidTaskDoneMessage();
+                    continue;
+                }
+                Task doneTask = taskList[doneTaskIndex - 1];
+                doneTask.markAsDone();
+
+                printTaskCompletedMessage(doneTask);
+            } else if (userInput.contains("todo")) {
+                // adds input as a "Todo" task to the task list
+                int todoPosition = userInput.indexOf("todo");
+                String todoTask = userInput.substring(todoPosition + 5);
+                Task newTaskToAdd = new Todo(todoTask);
+
+                taskList[taskCount] = newTaskToAdd;
+                taskCount++;
+
+                printTaskAddedMessage(taskCount, newTaskToAdd);
+            } else if (userInput.contains("deadline")) {
+                // adds input as a "Deadline" task to the task list
+                int deadlinePosition = userInput.indexOf("deadline");
+                int byPosition = userInput.indexOf("/by");
+                String deadlineTask = userInput.substring(deadlinePosition + 9, byPosition);
```

Comment:
Magic numbers 9 and 4! Perhaps you could make 9 and 4 a constant to DEADLINE_LENGTH and BY_LENGTH